File: clientapp/src/app/daily/dailyPayload.ts
import {Product} from "../service/payload/Product";
import {ProductAmountMealNumber} from "../service/payload/ProductAmountMealNumber";
import {RecipeMultiplierMealNumber} from "../service/payload/RecipeMultiplierMealNumber";

export class DailyPayload
{
  date: String;
  amountOfMeals: number;
  kcal: number;
  protein: number;
  fat: number;
  carbo: number;
  productAmountMealNumbers= new Array<ProductAmountMealNumber>();
  recipesMultiplierMealNumber= new  Array<RecipeMultiplierMealNumber>();
  excludedRecipesFromDayBefore= new Array<String>();
}

File: springbootapp/src/main/java/com/kosmala/springbootapp/payload/DailyPayload.java
package com.kosmala.springbootapp.payload;
import java.util.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class DailyPayload
{
    int amountOfMeals;
    int kcal;
    String date;
    int protein;
    int fat;
    int carbo;
    ////////////////////
    List<ProductAmountMealNumber> productAmountMealNumbers;
    List<RecipeMultiplierMealNumber> recipesMultiplierMealNumber;
    List<String> excludedRecipesFromDayBefore;
}
